diff --git a/swift-ci/sdks/static-linux/scripts/build.sh b/swift-ci/sdks/static-linux/scripts/build.sh
index 1ce2f3d..cbb9d68 100755
--- a/swift-ci/sdks/static-linux/scripts/build.sh
+++ b/swift-ci/sdks/static-linux/scripts/build.sh
@@ -179,32 +179,5 @@
 function describe {
-    pushd $1 >/dev/null 2>&1
-
-    desc=$(git describe --tags)
-    extra=
-
-    # Although git describe is documented as returning the newest tag,
-    # if a given revision has multiple tags, it will actually return
-    # the first one it finds, so we have some more work to do here.
-
-    # If we aren't pointing directly at a tag, git describe will append
-    # -<number-of-commits>-g<truncated-hash> to the nearest tag.  Strip
-    # this, but keep a note of it for later.
-    if [[ $desc =~ '-[0-9]+-g[0-9a-f]{11}$' ]]; then
-        stripped=${desc%-*-g*}
-        extra=${desc#$stripped}
-        desc=$stripped
-    fi
-
-    # Get the hash for the tag
-    rev=$(git rev-list -n 1 tags/$desc)
-
-    # Now find the newest tag at that hash, using version number ordering
-    latest_tag=$(git tag --points-at $rev | sort -V | tail -n 1)
-
-    # Stick it all back together
-    echo $latest_tag$extra
-
-    popd >/dev/null 2>&1
+    git -C "$1" describe --tags --always
 }
 function versionFromTag {
     desc=$(describe $1)
@@ -293,8 +266,8 @@ run cmake -G 'Ninja' ./ \
     -DCMAKE_USE_OPENSSL=OFF \
     -DBUILD_CursesDialog=OFF \
     -DBUILD_TESTING=OFF
-run ninja -C ${build_dir}/cmake/build
-run ninja -C ${build_dir}/cmake/build install
+run ninja -j1 -C ${build_dir}/cmake/build
+run ninja -j1 -C ${build_dir}/cmake/build install
 run export PATH="${build_dir}/cmake/install/bin:$PATH"
 quiet_popd
 run cmake --version
@@ -856,6 +829,6 @@
     SWIFT_SOURCE_ROOT="${source_dir}/swift-project" \
     SWIFT_BUILD_ROOT="${build_dir}/swift" \
-    run ${source_dir}/swift-project/swift/utils/build-script -r \
+    run ${source_dir}/swift-project/swift/utils/build-script -r --jobs=1 \
         --reconfigure \
         --compiler-vendor=apple \
         --bootstrapping hosttools \
